apps/showcase/src/components/training/code-editor-view/code-editor-view.component.ts

```diff
@@ -0,0 +1,159 @@
+export type EditorMode = 'readonly' | 'interactive';
+
+/** Project properties for the current exercise */
+export interface TrainingProject {
+  /** Starting file to be displayed in the project */
+  startingFile: string;
+  /** Files in the project */
+  files: FileSystemTree;
+  /** Commands to run in the project */
+  commands: string[];
+  /** Current working directory in project */
+  cwd: string;
+}
+
+@Component({
+  selector: 'code-editor-view',
+  standalone: true,
+  imports: [
+    AsyncPipe,
+    FormsModule,
+    MonacoEditorModule,
+    ReactiveFormsModule,
+    NgxMonacoTreeComponent,
+    CodeEditorControlComponent
+  ],
+  encapsulation: ViewEncapsulation.None,
+  templateUrl: './code-editor-view.component.html',
+  styleUrl: './code-editor-view.component.scss'
+})
+export class CodeEditorViewComponent implements OnDestroy, OnChanges {
+
+  @Input() public editorMode: EditorMode = 'readonly';
+  @Input() public project?: TrainingProject;
+
+  /** Variable to display "Log tree" button during debug mode */
+  public debugMode = false;
+
+  public readonly webContainerService = inject(WebContainerService);
+  public tree$ = this.webContainerService.monacoTree$;
+  private readonly formBuilder = inject(FormBuilder);
+  private readonly subscriptions = new Subscription();
+
+  public form: FormGroup = this.formBuilder.group({
+    code: '',
+    file: this.project?.startingFile || ''
+  });
+  public editorOptions$ = this.form.controls.file.valueChanges.pipe(
+    startWith(this.project?.startingFile || ''),
+    map((filePath: string) => ({
+      theme: 'vs-dark',
+      language: editorOptionsLanguage[filePath.split('.').pop() || 'ts'] || editorOptionsLanguage.ts,
+      readOnly: (this.editorMode === 'readonly'),
+      automaticLayout: true,
+      scrollBeyondLastLine: false
+    }))
+  );
+
+  constructor() {
+    this.subscriptions.add(
+      this.form.controls.code.valueChanges.pipe(
+        distinctUntilChanged(),
+        skip(1),
+        debounceTime(300)
+      ).subscribe(async (text) => {
+        if (!this.project) {
+          // eslint-disable-next-line no-console
+          console.error('No project found');
+          return;
+        }
+        const path = `${this.project.cwd}/${this.form.controls.file.value}`;
+        if (path && await this.webContainerService.isFile(path)) {
+          // eslint-disable-next-line no-console
+          console.log('Writing file', path);
+          void this.webContainerService.writeFile(`${this.project.cwd}/${this.form.controls.file.value}`, text);
+        }
+      })
+    );
+    this.subscriptions.add(
+      combineLatest([this.form.controls.file.valueChanges, this.webContainerService.isReady$])
+        .pipe(filter(([_, isReady]) => isReady))
+        .subscribe(async ([path, isReady]: [string, boolean]) => {
+          let content = '';
+          const isFile = isReady && this.project && await this.webContainerService.isFile(`${this.project.cwd}/${path}`);
```

Comment:
`isReady` is necessarily `true` due to the filter before
```suggestion
          const isFile = this.project && await this.webContainerService.isFile(`${this.project.cwd}/${path}`);
```

Reply:
Indeed, this is a leftover